pkg/logger/logger.go

```diff
@@ -0,0 +1,119 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type loggerState byte
+
+const (
+	new loggerState = iota
+	started
+	ended
+)
+
+type Logger struct {
+	filename string
+	state    loggerState
```

Comment:
Storing the current state in the logger isn't particularly important, to be honest. The state will be recorded by the types responsible for transitions between those states and this logger only really needs to take care of logging the information it receives.